core/shared/src/main/scala/scalaz/zio/ZManaged.scala

```diff
@@ -73,6 +73,51 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
       }
     }
 
+  final def foldM[R1 <: R, E2, B](
+    failure: E => ZManaged[R1, E2, B],
+    success: A => ZManaged[R1, E2, B]
+  ): ZManaged[R1, E2, B] =
+    ZManaged[R1, E2, B] {
+      Ref.make[ZIO[R1, Nothing, Any]](IO.unit).map { finalizers =>
+        Reservation(
+          acquire = {
+            val direct =
```

Comment:
Let's make sure to apply the adjustments re `restore` from the other PR here as well